app/src/main/kotlin/no/nav/omsorgsdager/vedtak/infotrygd/OmsorgspengerInfotrygdRammevedtakGateway.kt

```diff
@@ -1,12 +1,152 @@
+                barnetsIdentitetsnummer = it.barn().identitetsnummer(),
+                barnetsFødselsdato = it.barn().fødselsdato()
+            )
+        }
+
+        val midlertidigAlene = rammevedtak.getArray("MidlertidigAleneOmOmsorgen").mapJSONObject().map {
+            MidlertidigAleneInfotrygdInnvilgetVedtak(
+                kilder = it.kilder(),
+                vedtatt = it.vedtatt(),
+                gyldigFraOgMed = it.periode().fom,
+                gyldigTilOgMed = it.periode().tom
+            )
+
+        }
+
+        return utvidetRett
+            .asSequence()
+            .plus(midlertidigAlene)
+            .toList()
+    }
+
+    private fun authorizationHeader() =
+        cachedAccessTokenClient.getAccessToken(hentRammevedtakFraInfotrygdScopes).asAuthoriationHeader()
+
+    override suspend fun check() =
+        Result.merge(
+            "OmsorgspengerInfotrygdRammevedtakGateway",
+            accessTokenCheck(),
+            pingOmsorgspengerInfotrygdRammevetakCheck()
+        )
+
+    private fun accessTokenCheck() = kotlin.runCatching {
+        accessTokenClient.getAccessToken(hentRammevedtakFraInfotrygdScopes).let {
+            (SignedJWT.parse(it.accessToken).jwtClaimsSet.getStringArrayClaim("roles")?.toList()
+                ?: emptyList()).contains("access_as_application")
+        }
+    }.fold(
+        onSuccess = {
+            when (it) {
+                true -> Healthy("AccessTokenCheck", "OK")
+                false -> UnHealthy("AccessTokenCheck", "Feil: Mangler rettigheter")
+            }
+        },
+        onFailure = { UnHealthy("AccessTokenCheck", "Feil: ${it.message}") }
+    )
+
+
+    private suspend fun pingOmsorgspengerInfotrygdRammevetakCheck() =
+        pingUrl.httpGet().awaitStringResponseResult().third.fold(
+            success = { Healthy("PingOmsorgspengerInfotrygdRammevetak", "OK: $it") },
+            failure = { UnHealthy("PingOmsorgspengerInfotrygdRammevetak", "Feil: ${it.message}") }
+        )
+
+    private companion object {
+        private fun JSONObject.getArray(key: String) = when (has(key) && get(key) is JSONArray) {
+            true -> getJSONArray(key)
+            false -> JSONArray()
+        }
+
+        private fun JSONObject.periode() = Periode(
+            fom = LocalDate.parse(getString("gyldigFraOgMed")),
+            tom = LocalDate.parse(getString("gyldigTilOgMed"))
+        )
+
+        private fun JSONObject.kilder() = getJSONArray("kilder")
+            .map { it as JSONObject }
+            .map {
+                Kilde(
+                    id = it.getString("id"),
+                    type = it.getString("type")
+                )
+            }.toSet()
+
+        private fun JSONObject.barn() = getJSONObject("barn")
+        private fun JSONObject.vedtatt(): LocalDate = LocalDate.parse(getString("vedtatt"))
+        private fun JSONObject.identitetsnummer(): Identitetsnummer = getString("id").somIdentitetsnummer()
```

Comment:
`id` her er vel kun et identitetsnummer om `type` er `PersonIdent´ - om `type` er `Fødselsdato` så vil nok denne linja feile. Så vi bør enten sjekke at det er rett type for å parse som identitetsnummer, eller gjør dette i en `runCatching` og defaulte til null onFailure